### PR TITLE
Fixes for misc gem download issues

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
- IMPORTANT: change `originuri` to `origin_uri` to match the naming scheme of the rest of gem.json
- fix issue where downloaded gems cannot be added because gem download status has changed
- fix issue where gem could not be disabled if a path to the gem.json was provided instead of the gem folder.  Remote gems that do not exist on disk do not have a folder, so project manager was sending the path to the gem.json which is not a folder
- fixed issue where downloaded gems with hashes did not match because repo_name was added to remote gem jsons programatically
- fixed file handle left open when comparing hash
- fixed `template_name_name` syntax error and removed unused shutil import
- fix issue where uninstalled gems were not removed from projects

### CODE REVIEW

**Review by AMZN-nggieber — COMMENTED** [review]
Everything looks good just need to ensure this case is covered:
1. open the gem catalog with a remote gem that is undownloaded
2. download that gem 
3. then uninstall it
(Note: don't leave the gem catalog)
Gem cart should show no changes after that.

**Comment by AMZN-alexpete** [comment]
> Everything looks good just need to ensure this case is covered:
> 
> 1. open the gem catalog with a remote gem that is undownloaded
> 2. download that gem
> 3. then uninstall it
>    (Note: don't leave the gem catalog)
>    Gem cart should show no changes after that.

Good catch, pushed a fix for that

**Review by AMZN-nggieber — APPROVED** [review]

File: scripts/o3de/o3de/repo.py [hunk 1]
```diff
@@ -220,7 +219,7 @@ def search_repo(manifest_json_data: dict,
         o3de_object_uris = manifest_json_data['template']
```
Comment (lumberyard-employee-dm) [inline scripts/o3de/o3de/repo.py:219]: Actually the wrong key is being looked up here. It should be `templates` with an `s`.
```suggestion
        o3de_object_uris = manifest_json_data['templates']
```

File: scripts/o3de/o3de/download.py [hunk 2]
```diff
@@ -48,25 +48,26 @@ def validate_downloaded_zip_sha256(download_uri_json_data: dict, download_zip_pa
...
+            del download_uri_json_data[key]
+        if key in unzipped_manifest_json_data:
+            del unzipped_manifest_json_data[key]
 
     sha256A = hashlib.sha256(json.dumps(download_uri_json_data, indent=4).encode('utf8')).hexdigest()
     sha256B = hashlib.sha256(json.dumps(unzipped_manifest_json_data, indent=4).encode('utf8')).hexdigest()
     if sha256A != sha256B:
-        logger.error('SECURITY VIOLATION: Downloaded manifest json does not match'
+        logger.error(f'SECURITY VIOLATION: Downloaded {manifest_json_name} does not match'
                      ' the advertised manifest json.')
```
Comment (lumberyard-employee-dm) [inline scripts/o3de/o3de/download.py:71]: Wouldn't it be quicker to just compare the strings at this point?

Comment (AMZN-alexpete) [reply scripts/o3de/o3de/download.py:71]: Yeah, it would be simpler, I'm not sure why a hash was used before. @AMZN-Phil ?

Comment (AMZN-Phil) [reply scripts/o3de/o3de/download.py]: No reason that I can think of not to just use the dictionary check here.

**Review by AMZN-Phil — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**